pypsa/plot.py

```diff
@@ -543,6 +548,61 @@ def directed_flow(n, flow, x=None, y=None, ax=None, geomap=True,
     return arrowcol
 
 
+def autogenerate_coordinates(network, assign=False, layouter=None):
+    """
+    Automatically generate bus coordinates for the network graph
+    according to a layouting function from `networkx <https://networkx.github.io/>`_.
+
+    Parameters
+    ----------
+    network : pypsa.Network
+    assign : bool, default False
+        Assign generated coordinates to the network bus coordinates
+        at ``network.buses[['x','y']]``.
+    layouter : networkx.drawing.layout function, default None
+        Layouting function from `networkx <https://networkx.github.io/>`_. See
+        `list <https://networkx.github.io/documentation/stable/reference/drawing.html#module-networkx.drawing.layout>`_
+        of available options. By default coordinates are determined for a
+        `planar layout <https://networkx.github.io/documentation/stable/reference/generated/networkx.drawing.layout.planar_layout.html#networkx.drawing.layout.planar_layout>`_ 
+        if the network graph is planar, otherwise for a 
+        `Kamada-Kawai layout <https://networkx.github.io/documentation/stable/reference/generated/networkx.drawing.layout.kamada_kawai_layout.html#networkx.drawing.layout.kamada_kawai_layout>`_. 
+
+    Returns
+    -------
+    coordinates : pd.DataFrame
+        DataFrame containing the generated coordinates with
+        buses as index and ['x', 'y'] as columns.
+
+    Examples
+    --------
+    >>> autogenerate_coordinates(network)
+    >>> autogenerate_coordinates(network, assign=True, layouter=nx.circle_layout)
+    """
+
+    G = network.graph()
+    
+    if layouter is None:
+        is_planar = nx.check_planarity(G)[0]
+        if is_planar:
+            layouter = nx.planar_layout
+        else:
+            layouter = nx.kamada_kawai_layout
+            
+    coordinates = pd.DataFrame(layouter(G)).T.rename({0: 'x', 1: 'y'}, axis=1)
+    
+    if assign:
+        network.buses[['x', 'y']] = coordinates
+    
+    return coordinates
+
+
+def _get_coordinates(network, layouter=None):
+    if layouter is not None or network.buses[['x', 'y']].isin([0]).all().all():
```

Comment:
if layouter is not None or network.buses[['x', 'y']].isin([np.nan, 0]).all().all():